lib/ddtrace/contrib/resque/resque_job.rb

```diff
@@ -32,7 +33,11 @@ def on_failure_shutdown_tracer(*_)
         end
 
         def shutdown_tracer_when_forked!
-          tracer.shutdown! if forked?
+          return unless forked?
+
+          # Shutdown the tracer and then wait for the writer to flush
+          tracer.shutdown!
+          tracer.writer.worker.join if tracer.writer.is_a?(Datadog::Writer)
```

Comment:
Is this change actually effective? I'm comparing the use of `tracer.shutdown!` and `tracer.writer.worker.join`, and they both functionally do the same thing, by joining the worker thread.

From a glance, it doesn't seem like this change should do anything at all, but if it does, I'd like to understand why. Any other insight to share @brettlangdon?

Reply:
hmm, not sure how I read over the fact that `worker.stop` also calls `worker.join`...

well, then we probably need to try and approach a different way then, and maybe at this moment when we call `tracer.shutdown!` is not at an appropriate time.

if you look at #803 I do move it about a bit, but that is also because I changed the instrumentation method a bit (patching instead of hooks/middleware)